pkg/vm/builtins/builtin_runner.go

```diff
@@ -29,3 +33,25 @@ func Runner(name starknetParser.Builtin) memory.BuiltinRunner {
 		panic("Unknown builtin")
 	}
 }
+
+// GetAllocatedInstances calculates the number of instances of given builtin for current step and builtin ratio.
+// Ratio parameter defines the ratio between the number of steps to the number of builtin instances. It means that this builtin is expected to be used once every eight steps of the execution
```

Comment:
do you know why these ratios are fixed values in the layouts ? it feels weird as they entierely depend on the content of the cairo file dont they?

Reply:
I dont really know where those values come from, as they are fixed in the layouts and there is no documentation on them. From what I understand they are just some estimations, on how often will the instances of the builtins appear during execution and just for some optimizations I suppose. But I have no clue why that exact values, maybe some Monte Carlo like exepriments were run by StarkWare or sth

Reply:
I'll ask SW team about it and we can add a comment later in any case